pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -276,7 +276,7 @@ func (r *Reconciler) updateNodeIps(ctx context.Context, cr *hubv1alpha1.Cluster)
 		// the current nodeIpList (nodeIpList contains list of externalIPs or internalIPs)
 		// if the nodeIP is no longer available, we update the cluster CR on controller cluster
 		if cr.Status.NodeIPs == nil || len(cr.Status.NodeIPs) == 0 ||
-			!validatenodeips(nodeIPs, cr.Status.NodeIPs) {
+			!validatenodeips(nodeIPs, cr.Status.NodeIPs) || !isValidNodeIpList(nodeIPs) {
 			log.Info("Mismatch in node IP", "IP in use", cr.Status.NodeIPs, "IP to be used", nodeIPs)
```

Comment:
why is this needed? validatenodeips function is already returning false if the elements of array are not same.